### DISCUSSION TITLE
Using Custom Fields in Computed Fields

### DISCUSSION THREAD

**Discussion by smk4664** [nautobot/nautobot#1284] [Show and tell]
I thought I would share something I found very interesting and hopefully useful for others.  In Nautobot, all my devices are configured with short names instead of the FQDN.  Each of my sites has its own domain, and I wanted the devices to inherit the domain name from the site and use it to form the FQDN.  

I created a custom field that applies to dcim.site. 
![image](https://user-images.githubusercontent.com/17092348/150035146-e04c85ef-95bd-4b07-834e-00f4e424198e.png)

I then use this Custom Field in the Computed Field rule on the device.

`{{ obj.name }}.{{ obj.site.cf.domain }}`
![image](https://user-images.githubusercontent.com/17092348/150035279-99e237be-d44f-4a5b-b42f-5d7166b43e0d.png)

Here is an example of it in use.

![image](https://user-images.githubusercontent.com/17092348/150035504-57c2810e-615b-45c3-9fb2-6024792228ae.png)
![image](https://user-images.githubusercontent.com/17092348/150035581-83d05cc3-d98e-4439-b561-43423e13c388.png)

**Comment by jvanderaa** [comment]
Thank you!
